app/src/main/java/org/wikipedia/donate/DonorHistoryActivity.kt

```diff
@@ -0,0 +1,170 @@
+
+        binding.donationInfoContainer.isVisible = viewModel.isDonor
+
+        binding.donorStatus.setOnClickListener {
+            showDonorStatusDialog()
+        }
+
+        binding.lastDonationContainer.setOnClickListener {
+            showLastDonatedDatePicker()
+        }
+
+        binding.recurringDonorCheckbox.isChecked = viewModel.isRecurringDonor
+        binding.recurringDonorContainer.setOnClickListener {
+            viewModel.isRecurringDonor = !binding.recurringDonorCheckbox.isChecked
+            binding.recurringDonorCheckbox.isChecked = viewModel.isRecurringDonor
+        }
+
+        binding.donateButton.setOnClickListener {
+            launchDonateDialog()
+        }
+
+        binding.experimentLink.setOnClickListener {
+            CustomTabsUtil.openInCustomTab(this, getString(R.string.contributions_dashboard_wiki_url))
+        }
+
+        binding.saveButton.setOnClickListener {
+            viewModel.saveDonorHistory()
+            setResult(RESULT_DONOR_HISTORY_SAVED)
+            finish()
+        }
+        updateDonorStatusText()
+        updateLastDonatedText()
+    }
+
+    private fun updateDonorStatusText() {
+        var donorStatusTextColor = R.attr.primary_color
+        val donorStatusText = if (!Prefs.hasDonorHistorySaved && viewModel.currentDonorStatus == -1) {
+            donorStatusTextColor = R.attr.placeholder_color
+            R.string.donor_history_update_donor_status_default
+        } else if (viewModel.isDonor) {
+            R.string.donor_history_update_donor_status_donor
+        } else {
+            R.string.donor_history_update_donor_status_not_a_donor
+        }
+        binding.donorStatus.text = getString(donorStatusText)
+        binding.donorStatus.setTextColor(ResourceUtil.getThemedColorStateList(this, donorStatusTextColor))
+        binding.donateButton.isVisible = viewModel.currentDonorStatus == 1 // Not a donor
+        binding.donationInfoContainer.isVisible = viewModel.isDonor
+    }
+
+    private fun updateLastDonatedText() {
+        binding.lastDonationDate.isVisible = viewModel.lastDonated != null
+        var lastDonatedTextColor = R.attr.primary_color
+        val lastDonatedText = if (viewModel.lastDonated == null) {
+            lastDonatedTextColor = R.attr.placeholder_color
+            R.string.donor_history_last_donated_hint
+        } else {
+            R.string.donor_history_last_donated
+        }
+        binding.lastDonationLabel.text = getString(lastDonatedText)
+        binding.lastDonationLabel.setTextColor(ResourceUtil.getThemedColorStateList(this, lastDonatedTextColor))
+        viewModel.lastDonated?.let {
+            binding.lastDonationDate.text = DateUtils.getRelativeTimeSpanString(
+                viewModel.dateTimeToMilli(it),
+                System.currentTimeMillis(),
+                DateUtils.DAY_IN_MILLIS
+            )
+        }
+        binding.recurringDonorCheckbox.isEnabled = viewModel.lastDonated != null
```

Comment:
Are we only making recurringDonorCheckbox enabled/disabled but not recurringDonorContainer? even though checkbox is disabled we can still click recurringDonorContainer which set the checkbox.

Reply:
also we are not updating the preference isRecurringDonor when we tap the recurringDonorCheckbox

Reply:
Good catch! All done 👍